execute/plugin_functions.go

```diff
@@ -227,15 +269,47 @@ func truncateObservation(
 	// If the encoded observation is too large, start filtering data.
 	for len(encodedObs) > maxSize {
```

Comment:
You're not using this for loop anymore due to the early exit when `observationFitsSize(observation, maxSize)` becomes true. Is that OK?

Reply:
That's not true. We iterate over all the chains one by one and remove only one commit from each. To iterate over the chains again we need to have the outer loop.